### PR TITLE
[ACA-3245] Fetch alfresco repo name from an API

### PR DESCRIPTION
**Pull request by arditdomi**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [X] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACA-3245

**What is the new behaviour?**
The repository name is fetched by an API and not from the application config

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/form/services/activiti-alfresco.service.ts [hunk 1]
```diff
@@ -55,14 +55,26 @@ export class ActivitiContentService {
     /**
      * Returns a list of all the repositories configured
      *
-     * @param accountId
-     * @param folderId
      */
-    getAlfrescoRepositories(tenantId: number, includeAccount: boolean): Observable<any> {
```
Comment (eromano) [inline lib/core/form/services/activiti-alfresco.service.ts:61]: will be better leave the old method with the option, this is a breaking change and honestly have two method just to change a parameter I don't like is better have the possibility to pass it has option includeAccounts

File: lib/core/form/services/activiti-alfresco.service.ts [hunk 2]
```diff
@@ -54,15 +54,12 @@ export class ActivitiContentService {
 
     /**
      * Returns a list of all the repositories configured
-     *
-     * @param accountId
-     * @param folderId
+     * @param includeAccounts
      */
-    getAlfrescoRepositories(tenantId: number, includeAccount: boolean): Observable<any> {
+    getAlfrescoRepositories(includeAccounts: boolean): Observable<any> {
```
Comment (eromano) [inline lib/core/form/services/activiti-alfresco.service.ts:59]: tenantId is missing

Comment (arditdomi) [reply lib/core/form/services/activiti-alfresco.service.ts:59]: I'm not sure if is needed. In all places that we use this function we pass tenantId = null. The api is returning the same if we don't pass anything as tenantId or if we pass only the one that the user is part of (if he is part of any tenant). The only place that tenantId could be useful is when you are an admin user and you can query by using tenantId but I think this is not a valid case in our apps.

Comment (arditdomi) [reply lib/core/form/services/activiti-alfresco.service.ts:59]: I can re-add it it's just a thought. Thnx @eromano

Comment (mauriziovitale) [reply lib/core/form/services/activiti-alfresco.service.ts:59]: it was not used. the UI was passing null

File: lib/core/form/services/activiti-alfresco.service.ts [hunk 3]
```diff
@@ -54,15 +54,14 @@ export class ActivitiContentService {
...
     /**
      * Returns a list of all the repositories configured
-     *
-     * @param accountId
-     * @param folderId
+     * @param includeAccounts
+     * @param tenantId
      */
-    getAlfrescoRepositories(tenantId: number, includeAccount: boolean): Observable<any> {
+    getAlfrescoRepositories(includeAccounts: boolean, tenantId?: number): Observable<any> {
```
[the remark this answers is not in the document]
Comment (mauriziovitale) [reply lib/core/form/services/activiti-alfresco.service.ts]: both of them needs to be optionals

File: lib/core/form/services/activiti-alfresco.service.ts [hunk 4]
```diff
@@ -54,15 +54,14 @@ export class ActivitiContentService {
...
     /**
      * Returns a list of all the repositories configured
-     *
-     * @param accountId
-     * @param folderId
+     * @param tenantId
+     * @param includeAccounts
      */
-    getAlfrescoRepositories(tenantId: number, includeAccount: boolean): Observable<any> {
+    getAlfrescoRepositories(tenantId?: number, includeAccounts?: boolean): Observable<any> {
```
Comment (eromano) [inline lib/core/form/services/activiti-alfresco.service.ts:60]: includeAccount no includeAccounts

**Pull request merged**